### PR TITLE
Add caliper to saxpy

### PR DESCRIPTION
**Pull request by rfhaque**

### CODE REVIEW

**Review by pearce8 — APPROVED** [review]

**Review by pearce8 — CHANGES_REQUESTED** [review]
Why is the CI run failing?

**Comment by rfhaque** [comment]
> Why is the CI run failing?

@pearce8 Fixed. The license comments had a # instead of forward slashes.

**Review by pearce8 — APPROVED** [review]

File: repo/saxpy/saxpy.cc [hunk 1]
```diff
@@ -110,28 +119,69 @@ int main(int argc, char** argv) {
...
+      adiak_init(&comm);
+      adiak_user();
+      adiak_uid();
+      adiak_launchdate();
+      adiak_executable();
+      adiak_executablepath();
+      adiak_libraries();
+      adiak_cmdline();
+      adiak_hostname();
+      adiak_clustername();
```
Comment (daboehme) [inline repo/saxpy/saxpy.cc:138]: Let's just use `adiak_collect_all()` for these

File: repo/saxpy/saxpy.cc [hunk 2]
```diff
@@ -110,28 +119,69 @@ int main(int argc, char** argv) {
...
+      adiak_namevalue("compiler", adiak_general, NULL, "%s", SAXPY_COMPILER_ID);
+      adiak_namevalue("compiler version", adiak_general, NULL, "%s", SAXPY_COMPILER_VERSION);
+      CALI_MARK_BEGIN("main");
+
+      adiak_namevalue("Problem", adiak_general, NULL, "%s", "standard");
+      CALI_MARK_BEGIN("problem");
```
Comment (daboehme) [inline repo/saxpy/saxpy.cc:144]: I think the `problem` region is redundant since it's essentially the same as `main`

**Pull request merged**